### ISSUE TITLE
`Handle Download` Block - file path is not assigned to the variable

### ISSUE THREAD

**Issue by xartuu** [AutomaApp/automa#1850]
**Describe the bug**
After using the `Handle download` block, when the file is already downloaded and the next block is started, the file path of the downloaded file is not assigned to the set variable.

**To Reproduce**
1. Use the `Handle download` block.
2. Check the `Wait for the file to be downloaded` setting and enter the variable name in `Assign to variable`.
3. Run the workflow and download the file from any site
4. Go into the workflow logs and see if the variable has assigned itself

**Expected behavior**
After using the `Handle download` block, the `{{variables.path}} variable should contain the path to the downloaded file.

**Screenshots**
<img width="314" alt="image" src="https://github.com/user-attachments/assets/b527b9ce-3f0f-4e8a-a0d2-fc4d74196234">
<img width="1415" alt="image" src="https://github.com/user-attachments/assets/0ce8fa71-2a41-4816-b0cf-19f3b8e50ec1">

**Desktop (please complete the following information):**
 - OS: `MacOS`
 - Browser: `Brave Browser`
 - Extension Version: `1.28.27`

**Additional context**

**Comment by pzw943969386** [comment]
The processing download module may not get the file path immediately, you can add a delay

**Comment by xartuu** [comment]
I added a delay and tested:
- the file is downloaded in 2 seconds
- module after 20 seconds does not return the path
- module changes the previously set variable and leaves it empty

<img width="1792" alt="image" src="https://github.com/user-attachments/assets/c4c49863-6e53-44ad-921a-bbac296b4e84" />

**Comment by pzw943969386** [comment]
![image](https://github.com/user-attachments/assets/1c4a0213-866c-4bb9-a490-c0fe755092be)
Delay use before processing downloads

**Comment by xartuu** [comment]
Hello, I don't want this issue to die. 
The problem is still not fixed and the feature is not working. Can you find someone willing to fix it?

**Comment by xartuu** [comment]
I don't know how to reproduce my situation, as it occurs on a business portal with limited access. Overall, I click on a button that is supposed to generate a report with data (in .xlsx Excel). From what I can tell, JS processes the data, as the site does not send any request, and then a window pops up to save the file.

Along with my comment, I poked around a bit in the Automa code and came to a way that restores the operation in my case. I would describe it as “processing nearest/future download”. In short if the `downloadId` is not specified it listens for the download and if there is no `filename` when the download is finished it checks it in the Chrome API.

https://github.com/xartuu/automa/commit/0ce148ba3dd4ecad5a27bb3b884cacbaefd60049

**Comment by xartuu** [comment]
@bingren-yd did you take a look at it^?

**Comment by bingren-yd** [comment]
> [@bingren-yd](https://github.com/bingren-yd) did you take a look at it^?

Yes, I apologize for the slight delay. I've reviewed the code you provided, and it looks correct.

There's just one more step that would be helpful. If you could create a PR request in the Automa project following the GitHub flow (and then I'll merge it into the main branch), it would be much appreciated. This way, you'll be automatically credited in the next changelog and release, as you were the one who discovered and resolved the issue.

@xartuu
